text/0097-log-data-model.md

```diff
@@ -0,0 +1,972 @@
+[semantic conventions for Attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions).
+
+## Example Log Records
+
+Below are examples that show one possible representation of log records in JSON.
+These are just examples to help understand the data model. Don’t treat the
+examples as _the_ way to represent this data model in JSON.
+
+This document does not define the actual encoding and format of the log record
+representation. Format definitions will be done in separate OTEPs (e.g the log
+records may be represented as msgpack, JSON, Protocol Buffer messages, etc).
+
+Example 1
+
+```javascript
+{
+  "Timestamp": 1586960586000, // JSON needs to make a decision about 
+                              // how to represent nanoseconds.
+  "Attributes": {
+    "http.status_code": 500,
+    "http.url": "http://example.com",
+    "my.custom.application.tag": "hello",
+  },
+  "Resource": {
+    "service.name": "donut_shop",
+    "service.version": "semver:2.0.0",
+    "k8s.pod.uid": "1138528c-c36e-11e9-a1a7-42010a800198",
+  },
+  "TraceId": "f4dbb3edd765f620", // this is a byte sequence 
+                                  // (hex-encoded in JSON)
+  "SpanId": "43222c2d51a7abe3",
+  "SeverityText": "INFO",
+  "SeverityNumber": 9,
+  "Body": "20200415T072306-0700 INFO I like donuts"
+}
+```
+
+Example 2
+
+```javascript
+{
+  "Timestamp": 1586960586000,
+  ...
+  "Body": {
+    "i": "am",
+    "an": "event",
+    "of": {
+      "some": "complexity"
+    }
+  }
+}
+```
+
+## Open Questions
+
+- Should we store entire
+  [W3C Trace Context](https://www.w3.org/TR/trace-context/), including
+  `traceparent` and `tracestate` fields instead of only `TraceFlags`?
+  
+- Is `SeverityText`/`SeverityNumber` fields design good enough?
+
+- Early draft of this proposal specified that `Timestamp` should be populated
+  from a monotonic, NTP-synchronized source. I removed this requirement to avoid
+  confusion. Do we need any requirements for timestamp sources?
+
+- Is there a need for special treatment of security logs?
+
+## Alternate Design
+
+An
+[alternate design](https://docs.google.com/document/d/1ix9_4TQO3o-qyeyNhcOmqAc1MTyr-wnXxxsdWgCMn9c/edit?ts=5e990fe2#heading=h.cw69q2ga62p6)
+that used an envelop approach was considered but I did not find it to be overall
+better than this one.
+
+## Prior Art
+
+### RFC5424 Syslog
+
+[RFC5424](https://tools.ietf.org/html/rfc5424) defines structured log data
+format and protocol. The protocol is ubiquitous (although unfortunately many
+implementations don’t follow structured data recommendations). Here are some
+drawbacks that do not make Syslog a serious contender for a data model:
+
+- While it allows structured attributes the body of the message can be only a
+  string.
+
+- Severity is hard-coded to 8 possible numeric values, and does not allow custom
+  severity texts.
+
+- Structured data does not allow arbitrary nesting and is 2-level only.
```

Comment:
It allows arbitrary nesting within OID and name, so we can have:

```
[user@32473 id="123123"]
[meta.user@32473 id="123123"]
[data@32473.1 id="123123"]
```

Or any compilation of the above

Reply:
Replied to Syslog-related proposal elsewhere, resolving this.